### PR TITLE
Add Architecture::Sparc32Plus

### PR DESCRIPTION
**Pull request by glaubitz**
This adds support for 32-bit SPARC using the V9 baseline (V8plus).

### CODE REVIEW

File: src/write/elf/object.rs [hunk 1]
```diff
@@ -640,6 +641,7 @@ impl<'a> Object<'a> {
             (Architecture::Sbf, None) => elf::EM_SBF,
             (Architecture::Sharc, None) => elf::EM_SHARC,
             (Architecture::Sparc, None) => elf::EM_SPARC,
+            (Architecture::Sparc32Plus, None) => elf::EM_SPARC,
```
Comment (philipc) [inline src/write/elf/object.rs:644]:
```suggestion
            (Architecture::Sparc32Plus, None) => elf::EM_SPARC32PLUS,
```

You'll need this for the tests to pass. I'm surprised this didn't cause a problem for your use too, unless the linker allows combining objects with both machine types?

Comment (glaubitz) [reply src/write/elf/object.rs]: Apologies, this was an oversight and a result of a copy-and-paste error.

**Review by philipc — APPROVED** [review]
Thanks!

**Comment by glaubitz** [comment]
> Thanks!

Thanks for being a responsive upstream maintainer!

**Pull request merged**